frontend/src/screens/ProposedActivities.tsx

```diff
@@ -0,0 +1,343 @@
+import {
+  View,
+  Text,
+  TextInput,
+  StyleSheet,
+  TouchableWithoutFeedback,
+  ScrollView,
+  Alert,
+  Pressable,
+  Modal,
+} from "react-native";
+import { Entypo, MaterialCommunityIcons } from "@expo/vector-icons";
+import { material } from "react-native-typography";
+import React, { useState, useEffect } from "react";
+import GoogleMapInput from "@/components/GoogleMaps/GoogleMapInput";
+import { Trip } from "@/types";
+import DateTimePickerModal from "react-native-modal-datetime-picker";
+import { router } from "expo-router";
+import { EXPO_PUBLIC_HOST_URL, formatDateTime, tripDate } from "@/utils";
+import GooglePlacesInput from "@/components/GoogleMaps/GooglePlacesInput";
+
+interface props {
+  id: String;
+}
+
+export const ProposedActivities: React.FC<props> = (id: props) => {
+  const serverUrl = "http://" + EXPO_PUBLIC_HOST_URL + ":3000/";
+
+  //Declare useState
+  const [activityName, setActivityName] = useState("");
+  const [activityDescription, setActivityDescription] = useState("");
+  const [activityNote, setActivityNote] = useState("");
+  const [activityLocation, setActivityLocation] = useState({
+    address: "",
+    citystate: "",
+    latitude: 0,
+    longitude: 0,
+    radius: 0,
+  });
+
+  const [activityStartDate, setActivityStartDate] = useState(new Date());
+  const [activityEndDate, setActivityEndDate] = useState(new Date());
+  const [startDatePickerVisibility, setStartDatePickerVisibility] =
+    useState(false);
+  const [endDatePickerVisibility, setEndDatePickerVisibility] = useState(false);
+  const [isSaved, setIsSaved] = useState(false);
+  const [isFormFilled, setIsFormFilled] = useState(false);
+  const [isNoteSelected, setNoteSelected] = useState(false);
+  const [mapVisible, setMapVisible] = useState(false);
+  // setActivityStartDate(new Date(trip.startDate));
+  useEffect(() => {
+    const getTripData = async () => {
+      try {
+        const link = serverUrl + "trips/" + id.id;
+        const trip = await fetch(link);
+        const data = await trip.json();
+        setActivityLocation({ ...data.location });
+        setActivityStartDate(new Date(data.startDate));
+        setActivityEndDate(new Date(data.endDate));
+      } catch (error) {
+        console.log(error);
+      }
+    };
+
+    getTripData();
+  }, []);
+  //Header render + Return Button + Alert Pop up
+  const alertSuccess = () => {
+    Alert.alert("Success", "Good boy", [
```

Comment:
;;;-;;; consider changing to another message alert

Reply:
Huhu